src/ucp/core/ucp_request.h

```diff
@@ -251,6 +251,9 @@ struct ucp_recv_desc {
     uint32_t                payload_offset; /* Offset from end of the descriptor
                                              * to AM data */
     uint16_t                flags;          /* Flags */
+    int16_t                 priv_length;    /* Number of bytes consumed from
```

Comment:
is it actually "header_offset"?
want some consistent naming with payload_offset..
maybe it would help to add some ascii art here

Reply:
BTW would is it correct that we need this field only for releasing the descriptor?

Reply:
Yes, we need that only for descriptor release. Will add more comments plus art